webCommander/centralHub/xTwitter/twitterAPI.py

```diff
@@ -47,10 +50,8 @@ def request_to_x_api(endpoint, headers={}, params={}):
     response = x_session.get(endpoint, headers=headers, params=params)
 
     print(f'Request to API {response.status_code}')
-    while response.status_code == TOO_MANY_REQUESTS:
-        print('Too many requests, trying in 15 mins...')
-        sleep(900)
-        response = request_to_x_api(endpoint, headers={}, params=params)
+    if response.status_code == TOO_MANY_REQUESTS:
+        raise XAPIUsageExceeded(f"API Exceeded, try again in 15 Mins")
```

Comment:
you could extract 15 into a variable in case you would ever have to update it, it's now hardcoded in 3 places